test/gtest/miopendriver_conv2d_trans.cpp

```diff
@@ -0,0 +1,104 @@
+#include <gtest/gtest_common.hpp>
+
+#include <miopen/env.hpp>
+#include <miopen/miopen.h>
+#include <miopen/process.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_WITH_MIOPENDRIVER)
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPENDRIVER_MODE_CONV)
+
+namespace miopendriver_conv2d_trans {
+
+std::vector<std::string> GetTestCases()
+{
+    const std::string& modeConvolutionArg =
+        miopen::GetStringEnv(MIOPEN_ENV(MIOPENDRIVER_MODE_CONV));
+
+    // clang-format off
+    return std::vector<std::string>{
+        // Why we have to use the driver:
+        //   The transposed convolutions are paritally implemented in the convolution_api layer,
+        //   but test apps (including test_conv*) were designed as unit tests and, therefore, do not use the public API.
+        // Also serves as a regression test for https://github.com/ROCm/MIOpen/issues/2459.
+        {modeConvolutionArg + " -m trans -x 1 -y 1 -W 112 -H 112 -c 64 -n 8 -k 32 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -g 1 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 1 -y 7 -W 17 -H 17 -c 32 -n 128 -k 16 -p 3 -q 0 -u 1 -v 1 -l 1 -j 1 -g 2 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 10 -y 5 -W 341 -H 79 -c 32 -n 4 -k 8 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -g 4 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 20 -y 5 -W 700 -H 161 -c 1 -n 4 -k 32 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -g 1 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 3 -y 3 -W 108 -H 108 -c 3 -n 8 -k 64 -p 1 -q 1 -u 2 -v 2 -l 1 -j 1 -g 1 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 5 -y 5 -W 175 -H 40 -c 128 -n 16 -k 256 -p 1 -q 1 -u 2 -v 2 -l 1 -j 1 -g 1 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 5 -y 5 -W 700 -H 161 -c 1 -n 16 -k 64 -p 1 -q 1 -u 2 -v 2 -l 1 -j 1 -g 1 -F 0 -V 1"},
+        {modeConvolutionArg + " -m trans -x 7 -y 7 -W 224 -H 224 -c 3 -n 16 -k 64 -p 3 -q 3 -u 2 -v 2 -l 1 -j 1 -g 1 -F 0 -V 1"}
+    };
+    // clang-format on
+}
+
+using TestCase = decltype(GetTestCases())::value_type;
+
+class MIOpenDriverConv2dTransTest : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+bool IsTestSupportedForDevice()
+{
+    using namespace miopen::debug;
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx900>;
+    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
+}
+
+void RunMIOpenDriver()
+{
+    bool runTestSuite = miopen::IsEnabled(MIOPEN_ENV(MIOPEN_TEST_WITH_MIOPENDRIVER)) &&
+                        IsTestSupportedForDevice() &&
+                        miopen::IsEnabled(MIOPEN_ENV(MIOPEN_TEST_ALL)) &&
+                        (miopen::GetStringEnv(MIOPEN_ENV(MIOPEN_TEST_FLOAT_ARG)) == "--float" ||
+                         miopen::GetStringEnv(MIOPEN_ENV(MIOPEN_TEST_FLOAT_ARG)) == "--half" ||
+                         miopen::GetStringEnv(MIOPEN_ENV(MIOPEN_TEST_FLOAT_ARG)) == "--bf16");
+
```

Comment:
These tests won't be run in standalone mode, but we need this (the whole idea of switching to google test is to be able to run tests without CTest).

We should have three tests, one per each datatype. When `IsUnset(MIOPEN_ENV(MIOPEN_TEST_ALL)`, all three tests should be run (if `IsTestSupportedForDevice() == true`), see #1843.

Reply:
Since these call the driver, and it relies on the environment variable, we likely need to separate these into three tests, and then set the associated argument with each test. That way you can run them when TEST_ALL is not set, and otherwise only run them selectively when they are set. 

Does that sound correct?

Reply:
@BrianHarrisonAMD Thanks for quick feedback, Brian.

> Does that sound correct?

Yes, but let's see the PR ;)

Reply:
@BrianHarrisonAMD The fix in #3011 looks good 👍

Reply:
Thanks for the feedback!